jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/HandlerRegistry.java

```diff
@@ -25,40 +28,51 @@
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
-import java.util.function.Supplier;
 
-final class HandlerRegistry {
+public final class HandlerRegistry {
   private static final String INSTRUMENTATION_NAME = "io.opentelemetry.contrib.jfr";
   private static final String INSTRUMENTATION_VERSION = "1.7.0-SNAPSHOT";
 
   private final List<RecordedEventHandler> mappers;
 
-  private static final Map<String, List<Supplier<RecordedEventHandler>>> HANDLERS_PER_GC =
-      Map.of(
-          "G1",
-          List.of(G1HeapSummaryHandler::new),
-          "Parallel",
-          List.of(ParallelHeapSummaryHandler::new));
-
   private HandlerRegistry(List<? extends RecordedEventHandler> mappers) {
     this.mappers = new ArrayList<>(mappers);
   }
 
   static HandlerRegistry createDefault(MeterProvider meterProvider) {
+    HashSet<String> garbageCollectors = new HashSet<>();
+
     var handlers = new ArrayList<RecordedEventHandler>();
-    var seen = new HashSet<String>();
+    // Must gather all GC names before creating GC handlers that require the list of active GC names
     for (var bean : ManagementFactory.getGarbageCollectorMXBeans()) {
-      var name = bean.getName();
-      for (var gcType : HANDLERS_PER_GC.keySet()) {
-        if (name.contains(gcType)
-            && !seen.contains(gcType)
-            && HANDLERS_PER_GC.get(gcType) != null) {
-          handlers.addAll(HANDLERS_PER_GC.get(gcType).stream().map(s -> s.get()).toList());
-          seen.add(gcType);
-        }
+      garbageCollectors.add(bean.getName());
+    }
+
+    // Configure GC specific handlers
+    for (var name : garbageCollectors) {
+      switch (name) {
+        case "G1 Young Generation":
+          {
+            handlers.add(new G1HeapSummaryHandler());
+            handlers.add(new G1GarbageCollectionHandler());
+            break;
+          }
```

Comment:
#nit: probably can drop the curly braces here as they don't improve readability.